src/metricCollector.ts

```diff
@@ -59,14 +59,17 @@ export class MetricCollector {
       }
       this.logger.info('added queue', name);
       this.queuesByName.set(name, {
-        name,
-        queue: new Queue(name, {
-          ...this.bullOpts,
-          connection: this.defaultRedisClient,
-        }),
-        prefix: this.bullOpts.prefix || 'bull',
-        queueEvents: new QueueEvents(name)
-      });
+				name,
+				queue: new Queue(name, {
+					...this.bullOpts,
+					connection: this.defaultRedisClient,
+				}),
+				prefix: this.bullOpts.prefix || 'bull',
+				queueEvents: new QueueEvents(name, {
+					...this.bullOpts,
+					connection: new IoRedis(this.redisUri), // QueueEvents instances must not reuse Redis connections, see https://docs.bullmq.io/guide/connections
```

Comment:
let's move this next to defaultRedisClient in constructor as eventsRedisClient to avoid creating new instances on each addToQueueSet call